### PR TITLE
feat: Premium users can pick their theme for the leaderboards (fehmer)

### PR DESCRIPTION
**Pull request by fehmer**
### Description

Premium users can pick their profile theme which is used in the leader boards and on their profile page.

image

### CODE REVIEW

**Review by Miodec — COMMENTED** [review]
Wouldnt it make more sense to add this property on the `premium` object insetad on the user root?

File: frontend/src/html/popups.html [hunk 1]
```diff
@@ -1161,6 +1161,11 @@
       <label>badge</label>
       <div class="badgeSelectionContainer"></div>
     </div>
+    <div class="leaderboardTheme">
```
Comment (fehmer) [inline frontend/src/html/popups.html:1164]: hide if user has no premium flag

File: backend/src/dal/leaderboards.ts [hunk 2]
```diff
@@ -147,6 +147,9 @@ export async function update(
             [`${key}.consistency`]: {
               $ifNull: [`$${key}.consistency`, "$$REMOVE"],
             },
+            "user.leaderboardTheme": {
+              $ifNull: ["$inventory.leaderboardTheme", "$$REMOVE"],
```
Comment (fehmer) [inline backend/src/dal/leaderboards.ts:151]: - move to premium.leaderboardTheme
-  only set if premium is active

**Comment by fehmer** [comment]
- add to daily lb

File: frontend/src/ts/event-handlers/global.ts [hunk 3]
```diff
@@ -6,6 +6,7 @@ import { getCommandline } from "../utils/async-modules";
 
 document.addEventListener("keydown", async (e) => {
   if (PageTransition.get()) return;
+  if (e.key === undefined) return;
```
Comment (fehmer) [inline frontend/src/ts/event-handlers/global.ts:9]: using auto-fill on the login form fires this event and fails on `e.key.toLowerCase()`.

File: frontend/src/ts/ready.ts [hunk 4]
```diff
@@ -131,6 +131,7 @@ window.onunhandledrejection = function (e): void {
       customTitle: "DEV: Unhandled rejection",
       duration: 5,
     });
+    console.error(e);
```
Comment (fehmer) [inline frontend/src/ts/ready.ts:134]: On the notification we loose the call stack. Printing the error on the console lets us find the source of the error.

**Comment by monkeytypegeorge** [comment]
Continuous integration check(s) failed. Please review the failing check's logs and make the necessary changes. https://github.com/monkeytypegame/monkeytype/actions/runs/8343875829

File: frontend/src/ts/popups/quote-search-popup.ts [hunk 5]
```diff
@@ -382,7 +382,8 @@ $("#popups").on(
 
 $("#popups").on("click", "#quoteSearchPopup #goToApproveQuotes", () => {
   hide();
-  void QuoteApprovePopup.show(true);
+  //TODO: remove after master is fixed
```
Comment (fehmer) [inline frontend/src/ts/popups/quote-search-popup.ts:385]: remember to remove

**Pull request closed without merging**